django/cantusdb_project/main_app/views/source.py

```diff
@@ -218,6 +224,40 @@ def form_valid(self, form):
         return HttpResponseRedirect(self.get_success_url())
 
 
+class SourceDeleteView(LoginRequiredMixin, UserPassesTestMixin, DeleteView):
+    """The view for deleting a source object
+
+    This view is linked to in the source-edit page.
+    """
+
+    model = Source
+    template_name = "source_confirm_delete.html"
+
+    def test_func(self):
+        user = self.request.user
+        source_id = self.kwargs.get(self.pk_url_kwarg)
+        source = get_object_or_404(Source, id=source_id)
+
```

Comment:
You could just do `return user_can_edit_source(user, source)` here although the function name doesn't exactly match.

Reply:
good! I made this substitution, and the tests still pass.